express-notes-api/server.js

```diff
@@ -0,0 +1,117 @@
+import express from 'express';
+import fs from 'fs/promises';
+import { errorMiddleware } from './index.js';
+import { ClientError } from './client-error.js';
+
+const PORT = 8100;
+const app = express();
+
+// Read the JSON file
+const jsonData = await fs.readFile('data.json', 'utf8');
+
+// Parse the JSON data
+const data = JSON.parse(jsonData);
+
+app.listen(PORT, () => {
+  console.log('Listening on ' + PORT);
+});
```

Comment:
It's generally a better idea to start listening after the whole server has been created.